### ISSUE TITLE
Implement RECLIP

### ISSUE THREAD

**Issue by AmericanPresidentJimmyCarter** [mlfoundations/open_clip#490]
https://arxiv.org/abs/2304.06028

> RECLIP: Resource-efficient CLIP by Training with Small Images
> [Runze Li](https://arxiv.org/search/cs?searchtype=author&query=Li%2C+R), [Dahun Kim](https://arxiv.org/search/cs?searchtype=author&query=Kim%2C+D), [Bir Bhanu](https://arxiv.org/search/cs?searchtype=author&query=Bhanu%2C+B), [Weicheng Kuo](https://arxiv.org/search/cs?searchtype=author&query=Kuo%2C+W)

> We present RECLIP (Resource-efficient CLIP), a simple method that minimizes computational resource footprint for CLIP (Contrastive Language Image Pretraining). Inspired by the notion of coarse-to-fine in computer vision, we leverage small images to learn from large-scale language supervision efficiently, and finetune the model with high-resolution data in the end. Since the complexity of the vision transformer heavily depends on input image size, our approach significantly reduces the training resource requirements both in theory and in practice. Using the same batch size and training epoch, RECLIP achieves highly competitive zero-shot classification and image text retrieval accuracy with 6 to 8× less computational resources and 7 to 9× fewer FLOPs than the baseline. Compared to the state-of-the-art contrastive learning methods, RECLIP demonstrates 5 to 59× training resource savings while maintaining highly competitive zero-shot classification and retrieval performance. We hope this work will pave the path for the broader research community to explore language supervised pretraining in more resource-friendly settings.

Although they are slightly diminished in quality compared to normally trained CLIP models, they train in much less time. With RECLIP it may be possible to train a SoTA model with more parameters than ViT big G, since it will take significantly less compute.
